WS2012R2/lisa/remote-scripts/ica/vcpu_verify_online.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+########################################################################
+#
+# Linux on Hyper-V and Azure Test Code, ver. 1.0.0
+# Copyright (c) Microsoft Corporation
+#
+# All rights reserved. 
+# Licensed under the Apache License, Version 2.0 (the ""License"");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0  
+#
+# THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+# OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION
+# ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR
+# PURPOSE, MERCHANTABLITY OR NON-INFRINGEMENT.
+#
+# See the Apache Version 2.0 License for specific language governing
+# permissions and limitations under the License.
+#
+########################################################################
+
+########################################################################
+#
+# VCPU_verify_online.sh
+#
+# Description:
+#	This script was created to automate the testing of VCPU online or offline.
+#   This script will verify if all the CPUs can be offline by checking
+#	the /proc/cpuinfo file.
+#	The VM is configured with 4 CPU cores as part of the setup script,
+#	as each core can't be offline except vcpu0 for a successful test pass.
+#
+#	The test performs the following steps:
+#		1. Configures the VM with 4 cores (see the test case XML definition)	
+#		2. Make sure we have a constants.sh file.
+#		3. Looks for the Hyper-v timer property of each CPU under /proc/cpuinfo
+#		4. Verifies if each CPU can't be offline exinclude VCPU0.
+#     
+#	To pass test parameters into test cases, the host will create
+#	a file named constants.sh.  This file contains one or more
+#	variable definition.
+#
+# Note: The Host of Hyper-V 2012 R2 don't support the CPU online or offline, so
+# To make sure the cpu on guest can't be offline.
+################################################################
+
+ICA_TESTRUNNING="TestRunning"      # The test is running
+ICA_TESTCOMPLETED="TestCompleted"  # The test completed successfully
+ICA_TESTABORTED="TestAborted"      # Error during setup of test
+ICA_TESTFAILED="TestFailed"        # Error during execution of test
+
+CONSTANTS_FILE="constants.sh"
+nonCPU0inter=0
+
+LogMsg()
+{
+    echo `date "+%a %b %d %T %Y"` : ${1}    # To add the timestamp to the log file
+}
+
+UpdateTestState()
+{
+    echo $1 > $HOME/state.txt
+}
+
+#
+# Update LISA with the current status
+#
+cd ~
+UpdateTestState $ICA_TESTRUNNING
+LogMsg "Updating test case state to running"
+
+#
+# Source the constants file
+#
+if [ -e ./${CONSTANTS_FILE} ]; then
+    source ${CONSTANTS_FILE}
+else
+    msg="Error: no ${CONSTANTS_FILE} file"
+    LogMsg $msg
+    echo $msg >> ~/summary.log
+    UpdateTestState $ICA_TESTABORTED
+    exit 10
+fi
+
+if [ -e ~/summary.log ]; then
```

Comment:
this needs to be moved before sourcing constants, as there we write to summary already